columnflow/tasks/framework/mixins.py

```diff
@@ -502,6 +530,15 @@ def store_parts(self):
 
         return parts
 
+    def find_keep_columns(self: ConfigTask, collection: ColumnCollection) -> set[Route]:
+        if collection == ColumnCollection.ALL_FROM_PRODUCERS:
+            return set.union(*(
+                producer_inst.produced_columns
+                for producer_inst in self.producer_insts
+            ))
+
+        return super().find_keep_columns(collection)
```

Comment:
I think we also need this method for the `MLModelMixin` and `MLModelsMixin`

Reply:
Added :+1: